app/webpacker/components/CompetitionResultSubmission/UploadResultsJson.jsx

```diff
@@ -0,0 +1,42 @@
+import { useMutation } from '@tanstack/react-query';
+import React, { useState } from 'react';
+import { Form } from 'semantic-ui-react';
+import uploadResultsJson from './api/uploadResultsJson';
+import useCheckboxState from '../../lib/hooks/useCheckboxState';
+import Errored from '../Requests/Errored';
+
+export default function UploadResultsJson({ competitionId, isWrtViewing }) {
+  const [resultFile, setResultFile] = useState();
+  const [markResultSubmitted, setMarkResultSubmitted] = useCheckboxState(isWrtViewing);
+
+  const { mutate: uploadResultsJsonMutate, error, isError } = useMutation({
+    mutationFn: () => uploadResultsJson({ competitionId, resultFile, markResultSubmitted }),
+    onSuccess: () => {
+      window.location.reload();
```

Comment:
Why?

Reply:
(For context, I am not a fan of manual page reloads. The success hook should instead set some React state that then triggers a re-render with the updated "has been uploaded" information. Simplest case is a `useState` that is uesd as `hasUploaded`, so I want to understand why you think this isn't possible here)